src/java/org/apache/cassandra/cql3/restrictions/SingleColumnRestriction.java

```diff
@@ -873,6 +873,81 @@ public boolean isAnn()
         }
     }
 
+    /**
+     * A Bounded ANN Restriction is one that uses a distance as the limiting factor instead of a number of result
+     * vectors.
+     */
+    public static final class BoundedAnnRestriction extends SingleColumnRestriction
+    {
+        private final Term value;
+        private final Term distance;
+        private final boolean isInclusive;
+
+        public BoundedAnnRestriction(ColumnMetadata columnDef, Term value, Term distance, boolean isInclusive)
+        {
+            super(columnDef);
+            this.value = value;
+            this.distance = distance;
+            this.isInclusive = isInclusive;
+        }
+
+        public ByteBuffer value(QueryOptions options)
+        {
+            return value.bindAndGet(options);
+        }
+
+        @Override
+        public void addFunctionsTo(List<Function> functions)
+        {
+            value.addFunctionsTo(functions);
+            distance.addFunctionsTo(functions);
+        }
+
+        @Override
+        MultiColumnRestriction toMultiColumnRestriction()
+        {
+            throw new UnsupportedOperationException();
```

Comment:
// only used by partition and clustering restrictions

Reply:
I added your comment as a comment to the method.